packages/SwingSet/src/kernel/kernel.js

```diff
@@ -577,10 +578,12 @@ export default function buildKernel(kernelEndowments) {
    * error message for the problem.
    */
   function createVatDynamically(buildFnSrc) {
-    const endowments = { require: kernelRequire };
-    const buildFn = evaluateProgram(buildFnSrc, endowments);
+    const endowments = { require: kernelRequire, HandledPromise };
+    const nameSpace = evaluateProgram(buildFnSrc.source, endowments)();
+    const buildFn = () => harden({ start: nameSpace.makeContract.start });
```

Comment:
Please separate this use of `{ start: nameSpace.makeContract.start }` into the calling contractHost.  Then it can be as simple as:
```js
const buildFn = () => harden(nameSpace);
```

and the contractHost can do something like:
```
const bundle = ... get the contract bundle;
assert(bundle.moduleFormat === 'getExport');
const buildFnSrc = {
  ...bundle,
  source: `\
function getExport() {
  // This is an adapter from the contract's bundle to the format expected by createVatDynamically.
  const exports = (${bundle.source})();
  return { start: exports.makeContract.start };
}
`,
};
createVatDynamically(buildFnSrc);
```

Reply:
I think I understand where you're trying to go, but I don't see how to make it work.

In order to allow contracts to use import and tildot, they need to be processed, presumably by `bundleSource()`. But that means they'll already have `getExport() { ...}` wrapped around them. At that point, the start function can't easily be pulled out, since it's in the middle of a bunch of text, and we want the evaluated version anyway, since that has access to the imports and endowments.

I understand wanting to get rid of (or at least trim down) `namespace.makeContract.start`, but this doesn't seem to work.

Reply:
@michaelfig and I had an off-line conversation in which he clarified that the `getExport() in the code above is intended to wrap the `getExport()` that's already in the code. I'll call the new one `outerGetExport` to distinguish them. 

This does move the complexity out of the kernel and into ContractHost.